Modules/_threadmodule.c

```diff
@@ -1499,50 +1676,50 @@ unhandled exception; a stack trace will be printed unless the exception\n\
 is SystemExit.\n");
 
 static PyObject *
-thread_PyThread_start_joinable_thread(PyObject *module, PyObject *func)
+thread_PyThread_start_joinable_thread(PyObject *module, PyObject *fargs,
+                                      PyObject *fkwargs)
 {
+    static char *keywords[] = {"function", "daemon", NULL};
+    PyObject *func = NULL;
+    int daemon = 0;
     thread_module_state *state = get_thread_state(module);
+    if (!PyArg_ParseTupleAndKeywords(fargs, fkwargs,
+                                     "O|p:start_joinable_thread", keywords,
+                                     &func, &daemon)) {
+        return NULL;
+    }
 
     if (!PyCallable_Check(func)) {
         PyErr_SetString(PyExc_TypeError,
                         "thread function must be callable");
         return NULL;
     }
 
-    if (PySys_Audit("_thread.start_joinable_thread", "O", func) < 0) {
+    if (PySys_Audit("_thread.start_joinable_thread", "Oi", func, daemon) < 0) {
         return NULL;
     }
 
     PyObject* args = PyTuple_New(0);
     if (args == NULL) {
         return NULL;
     }
-    ThreadHandleObject* hobj = new_thread_handle(state);
-    if (hobj == NULL) {
-        Py_DECREF(args);
-        return NULL;
-    }
-    if (do_start_new_thread(state, func, args, /*kwargs=*/ NULL, /*joinable=*/ 1,
-                            &hobj->ident, &hobj->handle, hobj->thread_is_exiting)) {
-        Py_DECREF(args);
-        Py_DECREF(hobj);
-        return NULL;
-    }
-    set_thread_handle_state(hobj, THREAD_HANDLE_RUNNING);
+    PyObject *hobj = do_start_new_thread(state, func, args,
+                    /*kwargs=*/ NULL, daemon);
     Py_DECREF(args);
-    return (PyObject*) hobj;
+    return hobj;
 }
 
 PyDoc_STRVAR(start_joinable_doc,
-"start_joinable_thread(function)\n\
+"start_joinable_thread(function[, daemon]])\n\
 \n\
 *For internal use only*: start a new thread.\n\
 \n\
 Like start_new_thread(), this starts a new thread calling the given function.\n\
 Unlike start_new_thread(), this returns a handle object with methods to join\n\
 or detach the given thread.\n\
 This function is not for third-party code, please use the\n\
-`threading` module instead.\n");
+`threading` module instead. During finalization the runtime will not wait for\n\
+the thread to exit if daemon is truthy.\n");
```

Comment:
I think the language in the docs tends to be "if daemon is True" even when the parsing code coerces other values to `bool`.